proximal/tests/base_test.py

```diff
@@ -17,30 +18,24 @@ def assertItemsAlmostEqual(self,
         if type(b) is list:
             b = np.array(a)
 
-        vmax = np.max(
-            (np.linalg.norm(a.ravel(),
-                            np.Inf), np.linalg.norm(b.ravel(), np.Inf)))
+        assert a.shape == b.shape
 
-        rel_diff = np.linalg.norm(a.ravel() - b.ravel(), np.Inf)
+        norm_infinity_b = np.linalg.norm(b.ravel(), np.Inf)
 
-        if vmax == 0:
-            assert rel_diff < eps
+        if norm_infinity_b == 0:
+            # b is a zero matrix / vector
+            assert a == approx(b, abs=eps)
             return
 
-        rel_diff /= vmax
-
-        assert rel_diff < eps
+        assert a == approx(b, abs=norm_infinity_b * eps)
```

Comment:
I thought we should `assert a == approx(b, rel=eps)`, but somehow the relative difference `(a_i - b_i) / b_i` can exceed `eps` for a very small value of `b_i`.

Reply:
Ok, glad you've figured this out.